pkg/plugins/ifaces.go

```diff
@@ -70,3 +71,26 @@ type DataRequestHandler interface {
 	// HandleRequest handles a data request.
 	HandleRequest(context.Context, *models.DataSource, DataQuery) (DataResponse, error)
 }
+
+type PluginInstaller interface {
+	// Install finds the plugin given the provided information
+	// and installs in the provided plugins directory.
+	Install(pluginID, version, pluginsDirectory, pluginZipURL, pluginRepoURL string) error
+	// Uninstall removes the specified plugin from the provided plugins directory.
+	Uninstall(pluginID, pluginPath string) error
```

Comment:
why does this need the pluginPath?  if something exists with `pluginId`, we should know the path right?  Or is this trying to handle when you end up with multiple plugins with the same ID installed?

Reply:
Just because the CLI supports installing to a custom directory. Only really makes sense in the case of the CLI!

```
&cli.StringFlag{
    Name:    "pluginsDir",
    Usage:   "Path to the Grafana plugin directory",
    Value:   utils.GetGrafanaPluginDir(runtime.GOOS),
    EnvVars: []string{"GF_PLUGIN_DIR"},
}